app.rb

```diff
@@ -0,0 +1,88 @@
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  attr_reader :people, :books, :rentals
+
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def list_all_books
+    puts 'List of all books:'
+    books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_all_people
+    puts 'List of all people:'
+    people.each do |person|
+      puts "ID: #{person.id}, Name: #{person.name}"
+    end
+  end
+
+  def create_person(name, age, role, _classroom = nil)
+    if role == 'student'
+      person = Student.new(age, name: name)
+    elsif role == 'teacher'
+      puts 'Enter teacher specialization:'
+      specialization = gets.chomp
+      person = Teacher.new(age, specialization: specialization, name: name)
+    else
+      puts "Invalid role. Please choose 'student' or 'teacher'."
+      return
+    end
+
+    people << person
+    puts "Person created successfully. ID: #{person.id}, Name: #{person.name}"
+  end
+
+  def create_book(title, author)
+    book = Book.new(title, author)
+    books << book
+    puts "Book created successfully. Title: #{book.title}, Author: #{book.author}"
+  end
+
+  def create_rental(person_id, book_title)
```

Comment:
- [ ] When trying to create a rental it was asking for the book title. That way is a bit prone to error because someone might make a typing error which results in difficulties in achieving the desired outcome. Please make sure to display the list of books and ask the user to select a number as shown in the example video. 🙂️
- [ ] Creating a rental is throwing the following `ArgumentError`. Please fix it. 😎️
- [ ] I could not check the functionality of listing rentals because I could not create a rental in the first place. Please make sure that functionality is working as well. 🧐️

![Screenshot from 2023-07-01 06-19-27](https://github.com/Christelle-12/school_library/assets/17243325/29e30394-6b23-4fca-bd9a-0a4ca35ddf9d)